subprojects/core/src/main/java/org/gradle/groovy/scripts/internal/BuildScriptTransformer.java

```diff
@@ -53,6 +54,7 @@ public void register(CompilationUnit compilationUnit) {
         new StatementLabelsScriptTransformer().register(compilationUnit);
         new ModelBlockTransformer(scriptSource.getDisplayName(), scriptSource.getResource().getLocation().getURI()).register(compilationUnit);
         imperativeStatementDetectingTransformer.register(compilationUnit);
+        new CompoundAssignmentTransformer().register(compilationUnit);
```

Comment:
❔ This means that if a user writes a Gradle plugin in Groovy, the += on properties and collections would differently. What would be the difference in behavior? Should we forbid += in non-build script source code for these types to avoid silent misconfiguration?

Reply:
>This means that if a user writes a Gradle plugin in Groovy, the += on properties and collections would differently. What would be the difference in behavior? Should we forbid += in non-build script source code for these types to avoid silent misconfiguration?
 
With the current implementation it wouldn't work outside of build scripts at all, with the usual `missing plus()` method. It is solvable, but there are objections to have build script sugar in the "normal" code context.